DirectMap.h

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// -*- c++ -*-
+
+#ifndef FAISS_DIRECT_MAP_H
+#define FAISS_DIRECT_MAP_H
+
+/**
+ * Direct map: a way to map back from ids to inverted lists
```

Comment:
could you move this comment in front of the DirectMap object (better for doxygen)